Comprehension du dataset

In [2]:
import pandas as pd
import numpy as np

In [13]:
dataset = pd.read_csv('/Users/macbookpro/PycharmProjects/projet_credit_risk_modeling/data/processed/credit_risk_dataset_v0.csv')

dataset.head()

,person_age,person_income,person_home_ownership,person_emp_length,loan_intent,loan_grade,loan_amnt,loan_int_rate,loan_status,loan_percent_income,cb_person_default_on_file,cb_person_cred_hist_length
0,22,59000,RENT,123.0,PERSONAL,D,35000,16.02,1,0.59,Y,3
1,21,9600,OWN,5.0,EDUCATION,B,1000,11.14,0,0.10,N,2
2,25,9600,MORTGAGE,1.0,MEDICAL,C,5500,12.87,1,0.57,N,3
3,23,65500,RENT,4.0,MEDICAL,C,35000,15.23,1,0.53,N,2
4,24,54400,RENT,8.0,MEDICAL,C,35000,14.27,1,0.55,Y,4


In [14]:
dataset.select_dtypes(exclude='string').shape[1]

8

    - verification des types

In [11]:
dataset.info()

<class 'pandas.DataFrame'>
RangeIndex: 32581 entries, 0 to 32580
Data columns (total 12 columns):
 #   Column                      Non-Null Count  Dtype  
---  ------                      --------------  -----  
 0   person_age                  32581 non-null  int64  
 1   person_income               32581 non-null  int64  
 2   person_home_ownership       32581 non-null  str    
 3   person_emp_length           31686 non-null  float64
 4   loan_intent                 32581 non-null  str    
 5   loan_grade                  32581 non-null  str    
 6   loan_amnt                   32581 non-null  int64  
 7   loan_int_rate               29465 non-null  float64
 8   loan_status                 32581 non-null  int64  
 9   loan_percent_income         32581 non-null  float64
 10  cb_person_default_on_file   32581 non-null  str    
 11  cb_person_cred_hist_length  32581 non-null  int64  
dtypes: float64(3), int64(5), str(4)
memory usage: 3.0 MB


    - detection de valeur null

In [12]:
dataset.isnull().mean()

person_age                    0.000000
person_income                 0.000000
person_home_ownership         0.000000
person_emp_length             0.027470
loan_intent                   0.000000
loan_grade                    0.000000
loan_amnt                     0.000000
loan_int_rate                 0.095639
loan_status                   0.000000
loan_percent_income           0.000000
cb_person_default_on_file     0.000000
cb_person_cred_hist_length    0.000000
dtype: float64

- observation :

        - Le dataset compte 12 variable  et 32581 observation <br>
        - Le nombre de variable quantitative est de 8 et les variable categoriel sont 4 <br>
        - deux variables presente des valeurs null ( person_emp_length 2% , loan_int_rate  9%)




Dictionnaire

| features                  | types   | description                         | dependance                     |
|--------------------------|--------|-------------------------------------|--------------------------------|
| person_age               | int    | Age des individus                   | No                             |
| person_income            | int    | Revenu des individus                | No                             |
| person_home_ownership    | str    | Situation de domicile               | No                             |
| person_emp_length        | float  | Expérience de travail               | No                             |
| loan_intent              | str    | Intention de prêt                   | No                             |
| loan_grade               | str    | Note du prêt                        | No                             |
| loan_amnt                | int    | Montant du prêt                     | No                             |
| loan_int_rate            | float  | Taux d'intérêt du prêt              | No                             |
| loan_status              | binaire| Statut du prêt (défaut / non)       | No                             |
| loan_percent_income      | float  | % du prêt dans le revenu            | person_income & loan_amnt      |
| cb_person_default_on_file| str    | Historique de défaut                | No                             |
| cb_person_cred_hist_length| int   | Longueur de l'historique de crédit  | No                             |